src/status_im2/contexts/chat/composer/keyboard.cljs

```diff
@@ -20,18 +21,28 @@
       (async-storage/set-item! :kb-default-height (str height)))))
 
 (defn handle-emoji-kb-ios
-  "Opening emoji KB on iOS while maximized will cause a flicker up and down. This method handles that."
+  "Opening emoji KB on iOS will cause a flicker up and down due to height differences.
+  This method handles that by adding the extra difference between the keyboards. When the input is
+  expanded to a point where the added difference will make the composer go beyond the screen causing a flicker,
+  we're subtracting the difference so it only reaches the allowed max-height. We're not animating these
+  changes to make it appear seamless during transitions between keyboard types when maximized."
   [event
    {:keys [emoji-kb-extra-height]}
    {:keys [text-value]}
    {:keys [height saved-height]}
    {:keys [max-height]}]
-  (let [start-h         (oops/oget event "startCoordinates.height")
-        end-h           (oops/oget event "endCoordinates.height")
-        diff            (- end-h start-h)
-        max-height-diff (- max-height diff)
-        curr-text       @text-value]
-    (if (> (reanimated/get-shared-value height) max-height-diff)
+  (let [start-h                  (oops/oget event "startCoordinates.height")
+        end-h                    (oops/oget event "endCoordinates.height")
+        diff                     (- end-h start-h)
+        max-height-diff          (- max-height diff)
+        curr-text                @text-value
+        almost-expanded?         (> (reanimated/get-shared-value height) max-height-diff)
+        ;; NOTE: the `keyboardWillChangeFrame` is dispatched sometimes when blurring (the input)
+        ;; with a diff of 103, which we don't want to react to as it causes the input to increase.
+        ;; 60 is an arbitrary number to avoid that, but still include the emoji keyboard difference
+        ;; (with some buffer).
+        smaller-than-emoji-diff? (< diff 60)]
```

Comment:
what is `almost-expanded?` and `smaller-than-emoji-diff` ? I think there was no problem in `handle-emoji-kb-ios` ?

Reply:
`almost-expanded?` is the same condition we had before, just moved it inside `let`.  For `smaller-than-emoji-diff?` I tried to narrow down the condition for the emoji keyboard switch, as sometimes it would fire without toggling the emoji keyboard and as a result changing the animated height.

Reply:
@clauxx where did the value `60` come from in ` smaller-than-emoji-diff? (< diff 60)`

Reply:
@OmarBasem There's an inline comment above it that states it's an arbitrary value. It's somewhere in the middle between the emoji kb diff ~40 and the one we're avoiding 103.

Reply:
@clauxx the emoji kb diff will be different for different devices. Better not use such a value.

Reply:
That's why I left a buffer there of ~20. We could remove it and if it becomes an issue later figure out a better solution.